akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -501,7 +501,7 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
       seedNodes.mkString(", "),
       ShutdownAfterUnsuccessfulJoinSeedNodes)
     joinSeedNodesDeadline = None
-    CoordinatedShutdown(context.system).run(CoordinatedShutdown.ClusterDowningReason)
+    CoordinatedShutdown(context.system).run(CoordinatedShutdown.ClusterJoinUnsuccessfulReason)
```

Comment:
Currently we have `ClusterDowningReason` configured to use exit code `-1` (abnormal termination). Shouldn't ClusterJoinUnsuccessfulReason also be configured to use `-1`.

I guess -1 will have the effect that an orchestration platform will start it again, which is probably desired also for unsuccessful join?

Reply:
Hmmm, I think the exit `status` was relevant in ConductR but I don't think it is in Kubernetes. IDK about other orchestrators.

I'll fix this PR but I think we could consider deprecating the feature.

Reply:
The exit code is useful. k8 isn't the only thing. A bash script can check exit code for decision of restarting process.

Reply:
Hmmm, in the case of `IncompatibleConfigurationDetectedReason` restarting won't help so I don't think a custom exit code is necessary.

Reply:
I agree, IncompatibleConfigurationDetectedReason